api/test/org/apache/cloudstack/acl/RoleTypeTest.java

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.acl;
+
+import com.cloud.user.Account;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+
+public class RoleTypeTest {
+
+    @Test
+    public void testRoleTypeFromString() {
+        Assert.assertEquals(RoleType.fromString(null), null);
+        Assert.assertEquals(RoleType.fromString(""), null);
+        Assert.assertEquals(RoleType.fromString("admin"), null);
+        Assert.assertEquals(RoleType.fromString("12345%^&*"), null);
```

Comment:
Why doesn't ``fromString`` should either return ``Unknown`` or throw an ``IllegalStateException`` in these scenarios?